src/conductor/client/worker/worker_interface.py

```diff
@@ -42,6 +45,23 @@ def get_task_definition_name(self) -> str:
 
         :return: TaskResult
         """
+        return self.task_definition_name_cache
+
+    @property
+    def task_definition_name_cache(self):
+        if not hasattr(self, "_task_definition_name"):
+            self._task_definition_name_cache = self.compute_task_definition_name()
+        return self._task_definition_name_cache
+
+    def clear_task_definition_name_cache(self):
+        if hasattr(self, "_task_definition_name_cache"):
+            delattr(self, "_task_definition_name_cache")
+
+    def compute_task_definition_name(self):
+        if isinstance(self.task_definition_name, list):
+            task_definition_name = self.task_definition_name[self.next_task_index]
+            self.next_task_index = (self.next_task_index + 1) % len(self.task_definition_name)
```

Comment:
We should probably check for emptiness to make sure we don't run into modulo by 0 if `len(self.task_definition_name) == 0`.

Reply:
If someone instantiates a worker with an empty task list, would it be better to raise an error during initialization?